web/source/dom/input.ts

```diff
@@ -138,5 +142,25 @@ namespace com.keyman.dom {
       this.root.value = front + s + back;
       this.setCaret(caret + s._kmwLength());
     }
+
+    handleNewlineAtCaret(): void {
+      Input.newlineHandler(this.root);
+    }
+
+    static newlineHandler(inputEle: HTMLInputElement) {
+      // Can't occur for Mocks - just Input and TouchAlias types.
+      if (inputEle && (inputEle.type == 'search' || inputEle.type == 'submit')) {
```

Comment:
I don't understand this -- A `submit` input element is a button, so how would we have a keyboard active for it?